redisvl/extensions/router/semantic.py

```diff
@@ -40,6 +40,7 @@ class SemanticRouter(BaseModel):
     """The vectorizer used to embed route references."""
     routing_config: RoutingConfig = Field(default_factory=RoutingConfig)
     """Configuration for routing behavior."""
+    vector_field_name: str = "vector"
```

Comment:
Can we just use the reference to the `_index.schema` data in order to get to the field name here? Otherwise this will allow for a user to customize the vector field name, which we don't want

Reply:
I would like to keep things somewhat uniform across our cache, router, and session classes. `SemanticCache` and `SemanticSessionManager` both have this field. Should we remove it there also, or keep things different?

Reply:
That's a good point, I hadn't looked there.  Issue is with pydantic here. Because this is a `BaseModel` it will allow a user to set the vector_field_name param to be some str, but will default to `vector`. We don't want users to customize this variable. So if you can find a way to achieve that with pydantic or other, go for it!